apps/zotonic_mod_search/src/support/search_facet.erl

```diff
@@ -0,0 +1,351 @@
+%% @copyright 2021 Driebit BV
+%% @doc Faceted search using a facet.tpl for definition and a
+%% postgresql table for searches.
+
+%% Copyright 2021 Driebit BV
+%%
+%% Licensed under the Apache License, Version 2.0 (the "License");
+%% you may not use this file except in compliance with the License.
+%% You may obtain a copy of the License at
+%%
+%%     http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing, software
+%% distributed under the License is distributed on an "AS IS" BASIS,
+%% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+%% See the License for the specific language governing permissions and
+%% limitations under the License.
+
+-module(search_facet).
+
+-record(facet_def, {
+    name :: binary(),
+    block :: atom(),
+    type :: facet_type(),
+    is_range :: boolean()
+    }).
+
+-type facet_type() :: text
+                    | ft
+                    | int
+                    | float
+                    | dt
```

Comment:
Personally I prefer non-abbreviated names like datetime and fulltext. I had to read the rest of the code to realise I was slow on getting that dt means datetime ;)

Reply:
I made the names longer.